boards/components/src/bmi270.rs

```diff
@@ -0,0 +1,688 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2023.
+
+//! Component for the BMI270 Inertial Sensor.
+//!
+//! Usage
+//! -----
+//! ```rust
+//! let bmi270 = BMI270Component::new(mux_i2c, 0x68, mux_alarm).finalize(
+//!     components::bmi270_component_static!(nrf52840::rtc::Rtc<'static>, nrf52840::i2c::TWI));
+//! bmi270.begin_reset();
+//! let ninedof = components::ninedof::NineDofComponent::new(board_kernel)
+//!     .finalize(components::ninedof_component_static!(bmi270));
+//! ```
+
+use capsules_core::virtualizers::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use capsules_core::virtualizers::virtual_i2c::{I2CDevice, MuxI2C};
+use capsules_extra::bmi270::BMI270;
+use core::mem::MaybeUninit;
+use kernel::component::Component;
+use kernel::hil::i2c;
+use kernel::hil::time::Alarm;
+
+// Setup static space for the objects.
+#[macro_export]
+macro_rules! bmi270_component_static {
+    ($A:ty $(,)?, $I:ty) => {{
+        let i2c_device =
+            kernel::static_buf!(capsules_core::virtualizers::virtual_i2c::I2CDevice<$I>);
+        let alarm = kernel::static_buf!(
+            capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm<'static, $A>
+        );
+        let buffer = kernel::static_buf!([u8; 12]);
+        let config_file = kernel::static_buf!([u8; 8193]);
+        let bmi270 = kernel::static_buf!(
+            capsules_extra::bmi270::BMI270<
+                'static,
+                capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm<'static, $A>,
+                capsules_core::virtualizers::virtual_i2c::I2CDevice<$I>,
+            >
+        );
+
+        (i2c_device, alarm, buffer, config_file, bmi270)
+    };};
+}
+
+pub struct BMI270Component<A: 'static + Alarm<'static>, I: 'static + i2c::I2CMaster<'static>> {
```

Comment:
```suggestion
pub type BMI270ComponentType<A, I> = BMI270<VirtualMuxAlarm<A>, I2CDevice<I>>

pub struct BMI270Component<A: 'static + Alarm<'static>, I: 'static + i2c::I2CMaster<'static>> {
```

Please add the type for making the component easier to use.